Makefile

```diff
@@ -293,30 +317,96 @@ sycl-core.cpp  := $(sort $(wildcard backends/sycl/*.sycl.cpp))
 sycl-ref.cpp   := $(sort $(wildcard backends/sycl-ref/*.sycl.cpp))
 sycl-shared.cpp:= $(sort $(wildcard backends/sycl-shared/*.sycl.cpp))
 sycl-gen.cpp   := $(sort $(wildcard backends/sycl-gen/*.sycl.cpp))
+magma.c        := $(sort $(wildcard backends/magma/*.c))
+magma.cpp      := $(sort $(wildcard backends/magma/*.cpp))
+occa.cpp       := $(sort $(shell find backends/occa -type f -name *.cpp))
 
-hip-all.c := interface/ceed-hip.c $(hip.c) $(hip-ref.c) $(hip-shared.c) $(hip-gen.c)
+hip-all.c   := interface/ceed-hip.c $(hip.c) $(hip-ref.c) $(hip-shared.c) $(hip-gen.c)
```

Comment:
Any reason this is separate from the rest? Or why CUDA doesn't have an `all` source variable?

Reply:
That's a great question whose answer is unclear to me from the contents of the Makefile

Reply:
No reason, just style of the person writing it at the time. See where it's used (under `ifneq ($(CUDA_LIB_DIR),)` and `HIP_LIB_DIR`). They can be made consistent (either adding `cuda-all.c` or deleting `hip-all.c`).

Reply:
Ok, using *-all for both now